File: factorial.go
/*
 * Factorial Program in Go
 *
 * Compile:
 *   go build factorial.go
 *
 * Run:
 *   ./factorial
 *
 * Or directly:
 *   go run factorial.go
 */

package main

import "fmt"

func factorial1(numIn int) int {
	fac := 1
	counter := numIn

	if numIn > 0 {
		for counter > 1 {
			fac = fac * counter
			counter = counter - 1
		}
		return fac
	} else {
		return -1
	}
}

func main() {
	var input int

	fmt.Print("Enter the number >>> ")
	fmt.Scan(&input)

	if factorial1(input) == -1 {
		fmt.Println("Invalid Number in input !!!")
	} else {
		fmt.Printf("Factorial for %d is %d\n", input, factorial1(input))
	}
}

File: Factorial.scala
/*
 * Factorial Program in Scala
 *
 * Compile:
 *   scalac Factorial.scala
 *
 * Run:
 *   scala FactorialDriver
 *
 * Or with Scala 3:
 *   scala Factorial.scala
 */

def factorial1(numIn: Int): Int = {
  var fac = 1
  var counter = numIn

  if (numIn > 0) {
    while (counter > 1) {
      fac = fac * counter
      counter = counter - 1
    }
    fac
  } else {
    -1
  }
}

object FactorialDriver {
  def main(args: Array[String]): Unit = {
    print("Enter the number >>> ")
    val input = scala.io.StdIn.readInt()

    if (factorial1(input) == -1) {
      println("Invalid Number in input !!!")
    } else {
      println(s"Factorial for $input is ${factorial1(input)}")
    }
  }
}
